airflow/dags/monitor/test_retrieval.py

```diff
@@ -0,0 +1,267 @@
+
+
+@dag(
+    schedule_interval=None,
+    start_date=datetime(2023, 9, 27),
+    catchup=False,
+    is_paused_upon_creation=True,
+    default_args=default_args,
+    params={
+        "question_number_subset": Param(
+            default="",
+            title="Subset of test numbers to run.",
+            description=dedent(
+                """
+                A set of test ID numbers to run instead of the entire test set.
+                Format should be a bracketed list of integers
+                (ie. [14, 19, 21, 29, 32, 38, 50, 52])."""
+            ),
+        )
+    },
+)
+def test_retrieval(question_number_subset: str):
+    """
+    This DAG performs a test of document retrieval from Ask Astro's vector database.
+
+    It downloads a set of test questions as a CSV file from Google Sheets.
+    Set TEST_QUESTIONS_SHEET_ID environment variable to specify the source of test questions.
+
+    Retrieved answers and references are saved as CSV and uploaded to a Google Spreadsheet.
+    """
+
+    @task
+    def create_drive_folders(drive_folder: str) -> list:
+        """
+        This task creates Google Drive folders for the test results and returns the folder ID.
+        If the folders already exist it returns the folder ID.
+
+        :param drive_folder:  A fully-qualified path name for the folders. ie. "ask_astro/tests_dev"
+        :return: A list of folder IDs.
+        """
+
+        gd_hook = GoogleDriveHook(gcp_conn_id=_DRIVE_CONN_ID)
+
+        drive_folder_parts = drive_folder.split("/")
+
+        parent_id = get_or_create_drive_folder(gd_hook=gd_hook, folder_name=drive_folder_parts[0], parent_id=None)
+
+        folder_id = get_or_create_drive_folder(gd_hook=gd_hook, folder_name=drive_folder_parts[1], parent_id=parent_id)
+
+        gd_hook.get_conn().permissions().create(
+            fileId=parent_id, body={"type": "domain", "domain": google_domain_id, "role": "writer"}
+        ).execute()
+
+        return folder_id
+
+    @task
+    def get_schema(schema_file: str = "include/data/schema.json") -> list:
+        """
+        Get the schema object for this DAG.
+        """
+
+        class_objects = json.loads(Path(schema_file).read_text())
+        class_objects["classes"][0].update({"class": WEAVIATE_CLASS})
+
+        if "classes" not in class_objects:
+            class_objects = [class_objects]
+        else:
+            class_objects = class_objects["classes"]
+
+        return class_objects
+
+    @task
+    def check_schema(class_objects: dict) -> bool:
+        """
+        Check if the current schema includes the requested schema.  The current schema could be a superset
+        so check_schema_subset is used recursively to check that all objects in the requested schema are
+        represented in the current schema.
+        """
+
+        if AskAstroWeaviateHook(_WEAVIATE_CONN_ID).check_schema(class_objects=class_objects):
```

Comment:
wondering if we can directly use OSS provider here rather calling our wrapper

Reply:
Not yet.  Check_schema is not yet in OSS.  I could implement it here in DAG code but then this DAG is different from all the others in the repo.  Better to change them all to OSS when ready.